### PR TITLE
Redesign User Interface

### PR DESCRIPTION
**Pull request by Joanna-YJA**
This pull request redesigns the User Interface so that the user input text field is at the bottom and command result display is at the bottom. During the display of each type of Entity List as a result of commands, the Entity type will be highlighted on the left and badge which contains ID will be shown inside each Entity Card. Additionally, generics is used to implement Entity List Panel, do advise on whether it is alright, thank you!

## Checks
- [] No errors when running the tests
- [] Build and checkstyle passes
- [] Run the app and ran 2 commands without failing
- [] Written the baseline test cases for the PR

## Changelog
- []
- []
- []

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/addcommandparser/AddTeamCommandParser.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package seedu.address.logic.parser.addcommandparser;
 
+import java.util.logging.Logger;
```
Comment (john0227) [inline src/main/java/seedu/address/logic/parser/addcommandparser/AddTeamCommandParser.java]: Gotta reorder them imports.

File: src/main/java/seedu/address/ui/EntityCard.java [hunk 2]
```diff
@@ -79,6 +90,8 @@ public EntityCard(Entity entity, int displayedIndex) {
      * @param entity type of Entity.
      */
     private void generateParticipantCard(Entity entity) {
+        this.idIcon.setImage(getImage("/images/student_icon.png"));
```
Comment (john0227) [inline src/main/java/seedu/address/ui/EntityCard.java]: Maybe store paths to images as constants?

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 3]
```diff
@@ -168,10 +174,57 @@ private void handleExit() {
...
         return listPanel;
     }
 
+    /**
+     * Disarms or resets all buttons so that a new command can be carried out.
+     * The new command will arm a new button.
+     */
+    private void disarmAllButton() {
+        //TODO: shorten this
+        //Any ideas on how to shorten this method?
```
Comment (john0227) [inline src/main/java/seedu/address/ui/MainWindow.java]: An array of buttons? idk tbh

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 4]
```diff
@@ -197,20 +250,24 @@ private CommandResult executeCommand(String commandText) throws CommandException
             //TODO: if the current panel is the one being changed, do not change the entityListPlaceholder
             switch (type) {
             case M:
-                listPanel = new MentorListPanel(logic.getFilteredMentorList());
-                entityListPanelPlaceholder.getChildren().set(0, listPanel.getRoot());
+                disarmAllButton();
```
Comment (john0227) [inline src/main/java/seedu/address/ui/MainWindow.java]: Maybe you can have a fireButton method
```
void fireButton(Button b) {
    disarmAllButton()
    b.arm()
    b.fire()
}
```

**Comment by john0227** [comment]
Do we get to see cool Alfred now??😀

File: src/main/java/seedu/address/logic/parser/addcommandparser/AddTeamCommandParser.java (hunk 1 above)
Comment (Joanna-YJA) [reply src/main/java/seedu/address/logic/parser/addcommandparser/AddTeamCommandParser.java]: hahaha ok!

File: src/main/java/seedu/address/ui/EntityCard.java (hunk 2 above)
Comment (Joanna-YJA) [reply src/main/java/seedu/address/ui/EntityCard.java]: ah ok!

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 4 above)
Comment (Joanna-YJA) [reply src/main/java/seedu/address/ui/MainWindow.java]: oooh yea, thanks!

File: src/main/java/seedu/address/ui/CommandBox.java [hunk 5]
```diff
@@ -21,13 +22,18 @@
     @FXML
     private TextField commandTextField;
 
+    @FXML
+    private Button sendButton;
+
+
```
Comment (hcwong) [inline src/main/java/seedu/address/ui/CommandBox.java]: Standardize One line space between sections

File: src/main/java/seedu/address/ui/CommandBox.java [hunk 6]
```diff
@@ -21,13 +22,18 @@
...
     public CommandBox(CommandExecutor commandExecutor) {
         super(FXML);
         this.commandExecutor = commandExecutor;
         // calls #setStyleToDefault() whenever there is a change to the text of the command box.
         commandTextField.textProperty().addListener((unused1, unused2, unused3) -> setStyleToDefault());
     }
 
+
```
Comment (hcwong) [inline src/main/java/seedu/address/ui/CommandBox.java]: Same as above

File: src/main/java/seedu/address/ui/EntityCard.java [hunk 7]
```diff
@@ -21,8 +27,11 @@
  */
 public class EntityCard extends UiPart<Region> {
 
+
```
Comment (hcwong) [inline src/main/java/seedu/address/ui/EntityCard.java]: Same

File: src/main/java/seedu/address/ui/EntityCard.java [hunk 8]
```diff
@@ -44,10 +53,12 @@
 
     @FXML
     private Label name; //can be team name, mentor name or participant name
-
-
+    @FXML
```
Comment (hcwong) [inline src/main/java/seedu/address/ui/EntityCard.java]: I think should standardize line in between or line not inbetween

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 9]
```diff
@@ -168,10 +174,57 @@ private void handleExit() {
...
+
+    }
+
+    @FXML
+    private void displayMentorList() {
+        listPanel = new EntityListPanel(logic.getFilteredMentorList());
+        entityListPanelPlaceholder.getChildren().set(0, listPanel.getRoot());
+        entityListPanelPlaceholder.setStyle("-fx-background-color: #17202a");
+        logger.info("Color of entity list placeholder is: " + entityListPanelPlaceholder.getStyle());
+
```
Comment (hcwong) [inline src/main/java/seedu/address/ui/MainWindow.java]: Extra line here

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 3 above)
Comment (hcwong) [reply src/main/java/seedu/address/ui/MainWindow.java]: Think he means store all the references to buttons in an array so you can just filter and map over to disarm

File: src/main/resources/view/EntityCard.css [hunk 10]
```diff
@@ -0,0 +1,12 @@
+.cardPane{
+    -fx-effect: dropshadow(gaussian, rgba(0, 0, 0, 0.3), 10, 0.5, 0.0, 0.0);
+}
+
+.index, .id{
```
Comment (hcwong) [inline src/main/resources/view/EntityCard.css]: Spacing between class name and curly brace

File: src/main/resources/view/MainWindow.fxml [hunk 11]
```diff
@@ -1,60 +1,153 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (hcwong) [inline src/main/resources/view/MainWindow.fxml]: Can consider refactoring out in the future, but okay for now

File: build.gradle [hunk 12]
```diff
@@ -69,6 +69,9 @@ dependencies {
     testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: jUnitVersion
 
     compile "org.mockito:mockito-core:2.+"
+    implementation "de.jensd:fontawesomefx-fontawesome:4.7.0-11"
```
Comment (brianyenna) [inline build.gradle]: Hey just wanted to check if this additional dependency on an external library is approved? Was this mentioned on the Slack channel?

Comment (hcwong) [reply build.gradle]: Actually what are the approved external libraries?

File: src/main/java/seedu/address/ui/entitylistpanel/EntityListPanel.java [hunk 13]
```diff
@@ -1,14 +1,110 @@
...
+    private PrefixType prefix;
+
+    @FXML
+    private ListView<Entity> entityListView;
+    @FXML
+    private VBox panelContainer;
+
+    //TODO: clarify
+    //Not sure whether these generics is advisable, please advise
+    public EntityListPanel(ObservableList<? extends Entity> entityList) {
```
Comment (brianyenna) [inline src/main/java/seedu/address/ui/entitylistpanel/EntityListPanel.java]: Hey I'm not too great with generics, but maybe this is the sourceof the problem u mentioned in the telegram chat? Would it be possible to just put ObservableList<Entity> instead?

File: build.gradle (hunk 12 above)
Comment (brianyenna) [reply build.gradle]: Yeah, I'm not sure about this either. I just remember our tutor mentioning that whenever we use external libraries we should get it approved by asking on Slack first

Comment (hcwong) [reply build.gradle]: ill ask in lecture tmrw

File: src/main/java/seedu/address/ui/entitylistpanel/EntityListPanel.java (hunk 13 above)
Comment (Joanna-YJA) [reply src/main/java/seedu/address/ui/entitylistpanel/EntityListPanel.java]: Hey Brian, I solved the issue, thank you for the help :)

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 3 above)
Comment (john0227) [reply src/main/java/seedu/address/ui/MainWindow.java]: haha yea like a Button[] of all the buttons

File: src/main/resources/view/MainWindow.fxml (hunk 11 above)
Comment (Joanna-YJA) [reply src/main/resources/view/MainWindow.fxml]: Hey, what do you mean? hahah :)

Comment (hcwong) [reply src/main/resources/view/MainWindow.fxml]: Big XML files are hard to read i guess, but dont have to break it up for now.

File: src/main/resources/view/EntityCard.css (hunk 10 above)
Comment (Joanna-YJA) [reply src/main/resources/view/EntityCard.css]: Deleted this hahah

File: build.gradle (hunk 12 above)
Comment (Joanna-YJA) [reply build.gradle]: oh noo, ok i'll check in lecture tomorrow.

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 9 above)
Comment (Joanna-YJA) [reply src/main/java/seedu/address/ui/MainWindow.java]: Ok, I'll change it, thanks!

**Pull request merged**